File: elapsedTimeLogger.js
const elapsedTimeLogger = (prefix) => {
  let startTimeThisStage = Date.now();
  const prefix = prefix ? `${prefix}: ` : "";
  return (message) => {
    const now = Date.now();
    const elapsed = now - startTimeThisStage;
    startTimeThisStage = now;
    console.log(`${prefix}${message} took ${elapsed} ms`);
  };
};

File: elapsedTimeLogger.py
import time
def elapsedTimeLogger(namespace:str):
    start_this_stage=time.time()
    prefix = namespace+": "
    def logElapsedTime(message):
        nonlocal start_this_stage
        now = time.time()
        elapsed = now - start_this_stage
        start_this_stage = now
        print(f"{prefix}{message} took {elapsed} s")
    return logElapsedTime